src/data/lex.rs

```diff
@@ -15,6 +15,18 @@ pub struct Span {
     pub end: u32,
 }
 
+impl Span {
+    pub fn len(&self) -> usize {
+        (self.end - self.start) as usize // TODO is this unsafe?
```

Comment:
I assume by 'unsafe' you mean 'can panic'? Rust usually uses unsafe to mean memory corruption and that sort of thing.

Reply:
Yeah, that's what I mean. Is this reasonable to do?

Reply:
Yeah, the implementation looks fine.